### PR TITLE
Add support for `Service` intents

### LINKED ISSUE #18: Add support for services
The logic for creating a launch `Intent` can be exactly the same for `Service` as it is for `Activity`.

However, unlike an `Activity`, there is no `getIntent` function to power an `extras` extension property. Instead, services pass the `Intent` to functions as needed, such as `IntentService.onHandleIntent(intent: Intent?)`.

Cleanest solution is probably something like this:

```kotlin
// Your Source
@Exercise(Extra("example", Int::class))
class MyService : IntentService() {
    init { setIntentRedelivery(true) } // Lets us ignore null intent case to simplify sample

    override fun onHandleIntent(intent: Intent) {
        Log.d("MyService", "Example extra: ${extras(intent).example}")
    }
}

// Generated
class MyServiceParams(
    private val intent: Intent
) {
    val example: Int
        get() = intent.extras.get("${intent.component.packageName}.example") as Int
}

fun MyService.extras(intent: Intent): MyServiceParams
    get() = MyServiceParams(intent)
```

We don't get the parameter as a `val` extension, but this is probably the best we can do.

### PR DESCRIPTION
**Pull request by cedrickcooke**
Closes #18

### CODE REVIEW

File: application/src/test/java/com/juul/exercise/ExerciseProcessorServiceTests.kt [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+                class NoExtrasService : IntentService("test") {
+                    override fun onHandleIntent(intent: Intent?) { TODO() }
+                }
+                """
+            )
+        )
+        assertThat(result.exitCode).isEqualTo(KotlinCompilation.ExitCode.OK)
+
+        val file = result.getGeneratedFile("NoExtrasServiceExercise.kt")
+        assertThat(file.readText().trim()).isEqualToKotlin(
```
[the remark this answers is not in the document]
Comment (cedrickcooke) [reply application/src/test/java/com/juul/exercise/ExerciseProcessorServiceTests.kt:32]: It's a custom function, so don't get your hopes up. 😆 . I've considered making it more lenient, but I haven't had time to actually implement it. At this point, this is all it is:

```kotlin
/** Wrapper around [AbstractStringAssert.isEqualTo] for Kotlin code literals to syntax highlight in Android Studio. */
fun AbstractStringAssert<*>.isEqualToKotlin(
    @Language("kotlin") source: String
): AbstractStringAssert<*> = this.isEqualTo(source.trimIndent())
```

The comment on there was the real value add of having this.

File: application/src/test/java/com/juul/exercise/ExerciseProcessorServiceTests.kt [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile.Companion.kotlin
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class ExerciseProcessorServiceTests : ExerciseProcessorTests() {
+
+    @Test
+    fun `test service generation with no extras`() {
+        val result = compile(
```
[the remark this answers is not in the document]
Comment (cedrickcooke) [reply application/src/test/java/com/juul/exercise/ExerciseProcessorServiceTests.kt:12]: It's a really cool tool [here](https://github.com/tschuchortdev/kotlin-compile-testing), one of those tiny libraries that supports the whole ecosystem. Room, dagger, and moshi all use it to test against kotlin sources.

File: application/src/test/java/com/juul/exercise/ExerciseProcessorServiceTests.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (twyatt) [reply application/src/test/java/com/juul/exercise/ExerciseProcessorServiceTests.kt:32]: Oh. I saw `compile` then saw this and my mind went wild on the possibilities.
Still cool custom function but not quite as elaborate as what I unfairly imagined. 😝

[the remark this answers is not in the document]
Comment (cedrickcooke) [reply application/src/test/java/com/juul/exercise/ExerciseProcessorServiceTests.kt:32]: Filed #25. I don't think comparing parse trees would handle import changes, but it should at least handle things like whitespace

**Pull request merged**